hail/src/main/scala/is/hail/io/fs/AzureStorageFS.scala

```diff
@@ -212,10 +214,12 @@ class AzureStorageFS(val credentialsJSON: Option[String] = None) extends FS {
     }
   }
 
+  private lazy val httpClient = new NettyAsyncHttpClientBuilder().connectTimeout(Duration.ofSeconds(5)).build()
```

Comment:
Let's use the AzureBlobServiceClientBuilder's "ClientOptions" interface for this instead. That way, we can still let Azure pick the best defaults for its HttpClient. We want to set an [`HTTPClientOptions`](https://learn.microsoft.com/en-us/java/api/com.azure.core.util.httpclientoptions?view=azure-java-stable) for [`clientOptions`](https://learn.microsoft.com/en-us/java/api/com.azure.storage.blob.blobserviceclientbuilder?view=azure-java-stable#com-azure-storage-blob-blobserviceclientbuilder-clientoptions(com-azure-core-util-clientoptions))


e.g.

```
    val httpClientOptions = new HttpClientOptions()
    httpClientOptions.setReadTimeout(...)
    httpClientOptions.setResponseTimeout(...)
    httpClientOptions.setConnectTimeout(...)
    httpClientOptions.setConnectionIdleTimeout(...)
    httpClientOptions.setWriteTimeout(...)
    val blobServiceClient = new BlobServiceClientBuilder()
      .endpoint(s"https://${url.account}.blob.core.windows.net")
      .clientOptions(httpClientOptions)
      .buildClient()
```